src/h3lib/lib/geoCoord.c

```diff
@@ -136,47 +136,39 @@ double constrainLng(double lng) {
 }
 
 /**
- * Find the great circle distance in radians between two spherical coordinates.
+ * The great circle distance in radians between two spherical coordinates.
  *
- * @param p1 The first spherical coordinates.
- * @param p2 The second spherical coordinates.
- * @return The great circle distance in radians between p1 and p2.
+ * This function uses the Haversine formula.
+ * For math details, see:
+ *     https://en.wikipedia.org/wiki/Haversine_formula
+ *     https://www.movable-type.co.uk/scripts/latlong.html
+ *
+ * @param  a  the first lat/lng pair (in radians)
+ * @param  b  the second lat/lng pair (in radians)
+ *
+ * @return    the great circle distance in radians between a and b
  */
-double _geoDistRads(const GeoCoord* p1, const GeoCoord* p2) {
-    // use spherical triangle with p1 at A, p2 at B, and north pole at C
-    double bigC = fabs(p2->lon - p1->lon);
-    if (bigC > M_PI)  // assume we want the complement
-    {
-        // note that in this case they can't both be negative
-        double lon1 = p1->lon;
-        if (lon1 < 0.0L) lon1 += 2.0L * M_PI;
-        double lon2 = p2->lon;
-        if (lon2 < 0.0L) lon2 += 2.0L * M_PI;
+double H3_EXPORT(pointDistRads)(const GeoCoord *a, const GeoCoord *b) {
```

Comment:
Why is the implementation of this function replaced?

Reply:
A few reasons, but let me know if it would be better to make these changes and have the discussion in a separate PR.

1. According to https://www.movable-type.co.uk/scripts/latlong.html, the Haversine formula is more numerically stable than the law of cosines formula use previously. This doesn't seem to matter much for the examples I tested since we're using `double`s, but I figured it couldn't hurt to have us on sturdier numerical foundations.
2. I wanted to add links to further describe the algorithm. I couldn't find any references for the previous formula, but was able to find a few for the new formula.
3. Less branching logic.
4. Also according to https://www.movable-type.co.uk/scripts/latlong.html, the haversine formula may be faster, although I didn't see much difference in our tests. However, we could do some more rigorous benchmarking here.